lms/static/js/learner_dashboard/models/course_card_model.js

```diff
@@ -9,17 +9,34 @@
         function (Backbone) {
         return Backbone.Model.extend({
             initialize: function(data) {
+                this.activeRunMode = null;
                 if (data){
                     this.context = data;
                     //we should populate our model by looking at the run_modes
                     if (data.run_modes.length > 0){
-                        //We only have 1 run mode for this program
-                        this.setActiveRunMode(data.run_modes[0]);
+                        if(data.run_modes.length === 1){
+                            this.setActiveRunMode(data.run_modes[0]);
+                        }
+                        else{
+                            this.setActiveRunMode(
+                                this.getMostRecentRunMode(data.run_modes));
+                        }
```

Comment:
I would  move this if...else code into a more generic ```getRunMode``` function which would contain that logic.